src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -121,6 +169,21 @@ public override bool TrySetPropertyValue(string name, object value)
                 throw Error.ArgumentNull("name");
             }
 
+            if (_instanceAnnotationsPropertyInfo != null)
+            {                
+                if (name == _instanceAnnotationsPropertyInfo.Name)
+                {
+                    IODataInstanceAnnotationContainer annotationValue = value as IODataInstanceAnnotationContainer;
+                    if (_instanceAnnotationCache == null && annotationValue != null)
+                    {
+                        _instanceAnnotationCache =
+                            GetInstanceannotationContainer(_instanceAnnotationsPropertyInfo, _instance, annotationValue, create: true);
+                    }
+                                        
+                    return true;
```

Comment:
>true [](start = 27, length = 4)

should this return true if value is not set on line 179-180?

Reply:
I believe its still fine to return true, as the cache might be present